test/integration/tests/import_tpm.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+TPM_CC_DUPLICATE=0x14B
+TPM_CC_IMPORT=0x156
+
+source helpers.sh
+
+cleanup() {
+rm -f primary.ctx \
+          new_parent.prv new_parent.pub new_parent.ctx \
+          ipolicy.dat dpolicy.dat session.dat \
+          key.prv key.pub key.ctx \
+          dup.prv dup.pub dup.seed \
+          key2.prv key2.pub key2.ctx \
+          sym_key_in.bin
+
+    if [ "$1" != "no-shut-down" ]; then
+          shut_down
+    fi
+}
+trap cleanup EXIT
+
+start_up
+
+create_policy() {
+    tpm2_startauthsession -Q -S session.dat
+    tpm2_policycommandcode -Q -S session.dat -o $1 $2
+    tpm2_flushcontext -Q -S session.dat
+    rm session.dat
+}
+
+start_session() {
+    tpm2_startauthsession -Q --policy-session -S session.dat
+    tpm2_policycommandcode -Q -S session.dat -o $1 $2
+}
+
+end_session() {
+    tpm2_flushcontext -Q -S session.dat
+    rm session.dat
+}
+
+tpm2_createprimary -Q -a o -g sha256 -G rsa -o primary.ctx
+
+# Create new parent
+create_policy ipolicy.dat $TPM_CC_IMPORT
+tpm2_create -Q -C primary.ctx -g sha256 -G rsa -r new_parent.prv -u new_parent.pub -L ipolicy.dat -b "restricted|sensitivedataorigin|decrypt"
+# Load new parent key, only the public part
+tpm2_loadexternal -a o -u new_parent.pub -o new_parent.ctx
+
+# Create the key we want to duplicate
+create_policy dpolicy.dat $TPM_CC_DUPLICATE
+tpm2_create -Q -C primary.ctx -g sha256 -G rsa -r key.prv -u key.pub -L dpolicy.dat -b "sensitivedataorigin|decrypt"
+# Load the key
+tpm2_load -Q -C primary.ctx -r key.prv -u key.pub -o key.ctx
+
+# Extract the public part for import later
+tpm2_readpublic -c key.ctx -o dup.pub
+
+# Duplicate the key
+start_session dpolicy.dat $TPM_CC_DUPLICATE
+tpm2_duplicate -Q -C new_parent.ctx -c key.ctx -g null -p "session:session.dat" -r dup.dup -s dup.seed
+end_session
+
+rm new_parent.ctx
+
+# Load new parent key, public & private parts
+tpm2_load -Q -C primary.ctx -r new_parent.prv -u new_parent.pub -o new_parent.ctx
+
+# Import the duplicated key
+start_session ipolicy.dat $TPM_CC_IMPORT
+tpm2_import -V -C new_parent.ctx -g null -u dup.pub -d dup.dup -r dup.prv -s dup.seed -P "session:session.dat" -L dpolicy.dat
+end_session
+
+# Load imported key, public & private parts - HOW?
```

Comment:
Did you still have this question? -u and -r commands are for public and private portions respectively.

Reply:
Sorry, the comment is a bit cryptic. When I run the script as-is the final load fails. 

WARNING:esys:src/tss2-esys/api/Esys_Load.c:325:Esys_Load_Finish() Received TPM Error 
ERROR:esys:src/tss2-esys/api/Esys_Load.c:118:Esys_Load() Esys Finish ErrorCode (0x0000012f) 
ERROR: Eys_Load(0x12F) - tpm:error(2.0): authValue or authPolicy is not available for selected entity

I don't fully understand this but I think I need to satisfy the policy baked into new_parent?